qiskit/ignis/measurement/discriminator/iq_discrimination.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+import numpy as np
+from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
+from sklearn.discriminant_analysis import QuadraticDiscriminantAnalysis
+
+from qiskit.ignis.characterization.fitters import BaseFitter
+from qiskit.exceptions import QiskitError
+from qiskit.result.models import ExperimentResult
+from qiskit.result import postprocess
+
+
+class IQDiscriminationFitter(BaseFitter):
+
+    def __init__(self, cal_results, qubits, circuit_names, expected_states,
+                 discriminant,  description):
+        """
+        Args:
+            cal_results: calibration results, list of qiskit.Result or
+            qiskit.Result
+            qubits: the qubits for which we want to discriminate.
+            circuit_names: The names of the circuits in cal_results.
```

Comment:
This should be named `schedules` as circuits do not support measurement level 1 and should support passing the `Schedule` directly (the name string should also be acceptable). Also, this should be an optional list of schedules and should default to all experiments in`cal_results` if not provided.

Reply:
We can do schedules too. I take it we would then need to loop over the schedules and extract the name from each schedule and the name would then be the expected state?

Reply:
`get_memory` already does this for you. You can just pass the `Schedule` instance to it. We still require the expected state to be passed somewhere else.

Reply:
Got it. The newest push accounts for this now.